httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -175,6 +203,22 @@ public URIBuilder setCharset(final Charset charset) {
         return this;
     }
 
+    /**
+     * Sets the encoding policy for this {@link URIBuilder}.
+     * The encoding policy determines how URI components (e.g., query, fragment) are
+     * percent-encoded when building the URI string. If not set, the default policy
+     * is {@link EncodingPolicy#RFC_3986}.
+     *
+     * @param encodingPolicy the encoding policy to apply, or {@code null} to reset
+     *                       to the default ({@link EncodingPolicy#RFC_3986})
```

Comment:
I think this is no longer the case `ALL_RESERVED` seems to be the default now (see above).

Reply:
corrected